/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner_webconsole.proto;

message ParticipantExtra {
  reserved 2;
  bool is_manual_configured = 1;
}

// NextID: 12
message Participant {
  int64 id = 1;
  string name = 2;
  string domain_name = 3;
  string host = 4;
  int32 port = 5;
  // Ref to enum ParticipantType, this is the name.
  string type = 6;
  string comment = 7;
  int64 num_project = 8;
  // Timestamp in seconds
  int64 last_connected_at = 9;
  // Timestamp in seconds
  int64 created_at = 10;
  ParticipantExtra extra = 11;
  // Timestamp in seconds
  int64 updated_at = 12;
  string pure_domain_name = 13;
}
